google_permissions/pam_entitlement.tf

```
@@ -0,0 +1,266 @@
+  #  }
+  #
+  #  dynamic "approval_workflow" { //optional block
+  #    for_each = var.number_of_approvals > 0 ? [1] : []
+  #    content {
+  #      manual_approvals {
+  #        require_approver_justification =  each.value.nonprod.entitlement. # leave this false for now
+  #        steps {
+  #          approvals_needed          = 1 # this is all that's supported by google ATM
+  #          approver_email_recipients = []
+  #          approvers {
+  #            principals =
+  #          }
+  #        }
+  #      }
+  #    }
+  #  }
+}
+
+resource "google_privileged_access_manager_entitlement" "additional_entitlements" {
+  for_each = var.entitlement_enabled ? {
+    for entitlement in local.additional_entitlements : entitlement.key => entitlement
+  } : {}
+  entitlement_id       = each.value.entitlement.name
+  location             = "global"
+  max_request_duration = "${local.max_allowed_request_duration}s"
+  parent               = "projects/${each.value.project_id}"
+
+  requester_justification_config {
+    unstructured {}
+  }
+
+  eligible_users {
+    principals = local.module_outputs[each.key]
+  }
+  privileged_access {
+    gcp_iam_access {
+      dynamic "role_bindings" {
+        for_each = each.value.entitlement.roles
+        content {
+          role = role_bindings.value
+        }
+      }
+      resource      = "//cloudresourcemanager.googleapis.com/projects/${each.value.project_id}"
+      resource_type = "cloudresourcemanager.googleapis.com/Project"
+    }
+  }
+  dynamic "approval_workflow" { //optional block
+    for_each = try(length(each.value.entitlement.approval_workflow.principals), 0) > 0 ? [1] : []
+    content {
+      manual_approvals {
+        require_approver_justification = try(each.value.entitlement.approval_workflow.require_approver_justification, false) # leave this false for now
+        steps {
+          approvals_needed          = 1 # this is all that's supported by google ATM
+          approver_email_recipients = []
+          approvers {
+            principals = local.approvals_module_outputs[each.key]
+          }
+        }
+      }
+    }
+  }
+}
+
+resource "google_service_account" "account" {
+  for_each     = var.entitlement_enabled && var.entitlement_slack_topic != "" ? toset(local.environments) : []
+  account_id   = "slack-send-pam-sa"
+  display_name = "Slack sender function service account"
+  project      = local.environments[each.key]
+}
+
+
+# Create a feed that sends notifications about network resource updates.
+resource "google_cloud_asset_project_feed" "project_feed" {
+  for_each     = var.entitlement_enabled && var.entitlement_slack_topic != "" ? toset(local.environments) : []
+  project      = local.environments[each.key]
+  feed_id      = var.feed_id
+  content_type = "RESOURCE"
+
+  asset_types = [
+    "privilegedaccessmanager.googleapis.com/Grant",
+  ]
+
+  feed_output_config {
+    pubsub_destination {
+      topic = google_pubsub_topic.feed_output[each.key].id
```

```suggestion
      topic = google_pubsub_topic.feed_output[each.key].id
```

This should be variable, we should pass it in as part of the module.